demo/src/main/java/com/novoda/storagepathfinder/demo/AssetCloner.kt

```diff
@@ -0,0 +1,27 @@
+package com.novoda.storagepathfinder.demo
+
+import android.content.res.AssetManager
+import android.util.Log
+import java.io.File
+import java.io.FileOutputStream
+import java.io.IOException
+import java.util.concurrent.Executor
+
+internal class AssetCloner(private val assetManager: AssetManager, private val executor: Executor) {
+
+    fun cloneAsset(assetName: String, pathToCloneTo: String) {
+        executor.execute {
+            File(pathToCloneTo).also {
+                it.parentFile?.takeIf { !it.exists() }?.also { it.mkdirs() }
```

Comment:
Just want to make sure I understand this `?.` correctly, this says if the previous function didn't return `null` to perform the next section right? 

SO it checks for the parent file, if it isn't null then it checks that the parent `File` exists, if it does not then it creates the directories. After this it will then copy the assets to this location.

Reply:
Correct!